src/pesterTests.ts

```diff
@@ -53,6 +54,20 @@ export class PesterTestRunner {
 	
 		const exePath = await this.getPowerShellExe();
 
+		const pesterVersionScript = getPesterVersion();
+		this.log.debug(pesterVersionScript);
+		const pesterVersion = spawn(exePath, [
+			'-NonInteractive',
+			'-NoLogo',
+			'-NoProfile',
+			'-Command', pesterVersionScript]);
+		pesterVersion.stdout.on('data', (data) => {
+			const version = data.toString().replace(/[\n\r]+/g, '');
+			if ( (parseInt(version.split('.')[0]) == 5 && parseInt(version.split('.')[1]) < 2) || parseInt(version.split('.')[0]) < 5) {
```

Comment:
You're splitting the same string over and over again. Maybe do this instead

```ts
const [major, minor, patch] = version.split('.');
```
(this is called array destructuring if you've not seen it bofore, fun fact, PowerShell can do this too! `$major,$minor,$patch =...`)

Then do the `parseInt`s.

Reply:
Thank you, valid points .. it was late, I was tired.  I've come up with what I think is even a better solution, that involves passing JSON.

essentially:
`(Get-Module Pester).Version | ConvertTo-Json`

followed by:
`		pesterVersion.stdout.on('data', (data) => {
			const version = JSON.parse(data)
			if ( (parseInt(version.Major) == 5 && parseInt(version.Minor) < 2) || parseInt(version.Major) < 5) {
				vscode.window.showWarningMessage("Pester v5.2.0+ will be required for future versions of Pester Test Explorer.", 'OK');
			}
		});
`